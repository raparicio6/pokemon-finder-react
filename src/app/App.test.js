import React from 'react';
import { configure, mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';
import { Provider } from 'react-redux';

import App from './index';

configure({ adapter: new Adapter() });
describe('App', () => {
  it('renders without crashing', () => {
    const wrapper = mount(<App />);
    expect(wrapper.find(Provider)).toHaveLength(1);
  });
  it('matches the snapshot', () => {
    const tree = toJson(mount(<App />));
    expect(tree).toMatchSnapshot();
  });
});
