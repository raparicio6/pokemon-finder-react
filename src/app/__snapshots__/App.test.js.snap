// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App matches the snapshot 1`] = `
<App>
  <Provider
    store={
      Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      }
    }
  >
    <AppRoutes>
      <BrowserRouter>
        <Router
          history={
            Object {
              "action": "POP",
              "block": [Function],
              "createHref": [Function],
              "go": [Function],
              "goBack": [Function],
              "goForward": [Function],
              "length": 1,
              "listen": [Function],
              "location": Object {
                "hash": "",
                "pathname": "/",
                "search": "",
                "state": undefined,
              },
              "push": [Function],
              "replace": [Function],
            }
          }
        >
          <Switch>
            <AppRoute
              component={[Function]}
              computedMatch={
                Object {
                  "isExact": true,
                  "params": Object {},
                  "path": "/",
                  "url": "/",
                }
              }
              location={
                Object {
                  "hash": "",
                  "pathname": "/",
                  "search": "",
                  "state": undefined,
                }
              }
              path="/"
            >
              <Route
                computedMatch={
                  Object {
                    "isExact": true,
                    "params": Object {},
                    "path": "/",
                    "url": "/",
                  }
                }
                location={
                  Object {
                    "hash": "",
                    "pathname": "/",
                    "search": "",
                    "state": undefined,
                  }
                }
                path="/"
                render={[Function]}
              >
                <HomeContainer
                  history={
                    Object {
                      "action": "POP",
                      "block": [Function],
                      "createHref": [Function],
                      "go": [Function],
                      "goBack": [Function],
                      "goForward": [Function],
                      "length": 1,
                      "listen": [Function],
                      "location": Object {
                        "hash": "",
                        "pathname": "/",
                        "search": "",
                        "state": undefined,
                      },
                      "push": [Function],
                      "replace": [Function],
                    }
                  }
                  location={
                    Object {
                      "hash": "",
                      "pathname": "/",
                      "search": "",
                      "state": undefined,
                    }
                  }
                  match={
                    Object {
                      "isExact": true,
                      "params": Object {},
                      "path": "/",
                      "url": "/",
                    }
                  }
                >
                  <Home>
                    <div
                      className="languageButtons"
                    >
                      <LanguageButtonsContainer>
                        <LanguageButtons
                          onHandleLanguageChangeClick={[Function]}
                        >
                          <div
                            className="row"
                          >
                            <button
                              className="esButton"
                              data-lang="es"
                              onClick={[Function]}
                              type="button"
                            >
                              ES
                            </button>
                            <button
                              className="enButton"
                              data-lang="en"
                              onClick={[Function]}
                              type="button"
                            >
                              EN
                            </button>
                          </div>
                        </LanguageButtons>
                      </LanguageButtonsContainer>
                    </div>
                    <div
                      className="column center background-wild-sand all-screen-height container"
                    >
                      <h1
                        className="title m-bottom-6"
                      />
                      <h2
                        className="field m-bottom-6"
                      />
                      <SearchBarContainer>
                        <SearchBar
                          handleOnChange={[Function]}
                        >
                          <input
                            className="searchBar"
                            onChange={[Function]}
                            placeholder=""
                            type="search"
                          />
                        </SearchBar>
                      </SearchBarContainer>
                      <PokemonListContainer>
                        <PokemonList
                          pokemons={Array []}
                        >
                          <p
                            className="row center noPokemonsMessage m-top-16"
                          />
                        </PokemonList>
                      </PokemonListContainer>
                      <Footer>
                        <div
                          className="container row middle"
                        >
                          <span
                            className="information"
                          />
                          <a
                            className="repoLink"
                            href="https://github.com/raparicio6/pokemon-finder-react"
                          />
                        </div>
                      </Footer>
                    </div>
                  </Home>
                </HomeContainer>
              </Route>
            </AppRoute>
          </Switch>
        </Router>
      </BrowserRouter>
    </AppRoutes>
  </Provider>
</App>
`;
